### DISCUSSION TITLE
Id for Components?

### DISCUSSION THREAD

**Discussion by Tonaion02** [bevyengine/bevy#14570] [Q&A]
I can't understand if each Component has an Id in bevy, some sort of integer identifier for component types.

**Comment by SkiFire13** [comment]
Yes https://docs.rs/bevy/latest/bevy/ecs/component/struct.ComponentId.html

**Comment by PhantomMorrigan** [comment]
Yes, you can obtain it with `World::component_id`. Note that you can have multiple components types of the same rust type, or components made of non-rust types. The API for creating such component types is `World::init_component_with_descriptor`, but it should only really be used for modding.
